packages/react/src/components/Button/SplitButton.tsx

```diff
@@ -0,0 +1,155 @@
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import { Ref } from '@stardust-ui/react-component-ref'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import { WithAsProp, withSafeTypeForAs, ComponentEventHandler, ShorthandValue } from '../../types'
+import {
+  UIComponent,
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+  applyAccessibilityKeyHandlers,
+  commonPropTypes,
+} from '../../lib'
+import { Accessibility } from '../../lib/accessibility/types'
+import Button, { ButtonProps } from './Button'
+import MenuButton from '../MenuButton/MenuButton'
+import { splitButtonBehavior } from '../../lib/accessibility'
+import { MenuProps } from '../Menu/Menu'
+
+export interface SplitButtonProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   */
+  accessibility?: Accessibility
+
+  /** Element to be rendered in-place where the popup is defined. */
+  button?: ShorthandValue<ButtonProps>
+
+  /** Element to be rendered in-place where the popup is defined. */
+  menu?: ShorthandValue<MenuProps>
+
+  /**
+   * Called after user's click.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: ComponentEventHandler<ButtonProps>
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  primary?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  secondary?: boolean
+}
+
+class SplitButton extends UIComponent<WithAsProp<SplitButtonProps>> {
+  static create: Function
+
+  static displayName = 'SplitButton'
+
+  static className = 'ui-splitbutton'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    button: customPropTypes.itemShorthand,
+    menu: customPropTypes.itemShorthand,
+    onClick: PropTypes.func,
+    onToggle: PropTypes.func,
+    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
+  }
+
+  static defaultProps = {
+    accessibility: splitButtonBehavior,
+    as: 'span',
+    primary: true,
+  }
+
+  buttonRef = React.createRef<HTMLElement>()
+  menuButtonRef = React.createRef<MenuButton>()
+
+  actionHandlers = {
+    closeMenuAndFocusButton: () => {
+      this.closeMenu()
+      this.buttonRef.current.focus()
+    },
+    openAndFocusFirst: e => this.openAndFocusFirst(e),
+  }
+
+  closeMenu() {
+    this.menuButtonRef.current.closeMenu()
```

Comment:
We controlling the state of child component in the hidden way. In the current implementation there is no way to make `menu` controlled, i.e. there are no `menuOpen`, `onMenuChange` props.